shell/browser/web_dialog_helper.cc

```diff
@@ -171,8 +173,11 @@ class FileSelectHelper : public base::RefCounted<FileSelectHelper>,
             blink::mojom::NativeFileInfo::New(
                 path, path.BaseName().AsUTF16Unsafe())));
       }
+      // We should only call this if we have not cancelled the dialog
+      OnFilesSelected(std::move(file_info), base::FilePath());
+    } else {
+      OnSelectionCancelled();
```

Comment:
Same as above wrt removing negative logic now that we have actions for both the cancelled and not-cancelled state

```cc
if (cancelled) {
  OnSelectionCancelled();
} else {
  ...
}
```